### PR TITLE
Added Square Free Number in Mathematical Algorithms

### PR DESCRIPTION
**Pull request by mansiag**
**Fixes issue:**
N/A

**Changes:**
Added program to check whether a number is square free or not.

### CODE REVIEW

File: code/mathematical_algorithms/src/square_free_number/square_free_number.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//Program to check whether a number is square_free or not
+#include<stdio.h>
+#include<math.h>
+#include<stdlib.h>
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.c]: Add space before `<`.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+int isSquareFree(int n)
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.c]: Type goes on its own line.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+{
+	if(n % 2 == 0)
+		n = n / 2;
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.c]: Are you tabs are used?

Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.c]: `/=` operator.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	if(n % 2 == 0)
+		return 0;
+	int i;
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.c]: Fix indent.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    for( i = 3; i <= sqrt(n); i += 2)
+    {
+    	if(n % i == 0)
+    	{
+    		n = n / i;
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.c]: `/=`

File: code/mathematical_algorithms/src/square_free_number/square_free_number.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
+//Program to check whether a number is square_free or not
+#include<iostream>
+#include<cmath>
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.cpp]: Space before `<`.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.cpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+bool isSquareFree(int n)
+{
+	if(n % 2 == 0)
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.cpp]: Use 4 spaces here.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.cpp [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    {
+    	if(n % i == 0)
+    	{
+    		n = n / i;
+    		if(n % i == 0)
+    			return false;
+    	}
+    }
+    return true;
+}
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.cpp]: Same comments as C code.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.cpp [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+#include<iostream>
+#include<cmath>
+
+bool isSquareFree(int n)
+{
+	if(n % 2 == 0)
+		n = n / 2;
+	if(n % 2 == 0)
+		return false;
+    for(int i = 3; i <= sqrt(n); i += 2)
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.cpp]: `std::sqrt`.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
+// Program to check whether a number is square_free or not
+import java.util.Scanner;
+class square_free_number{
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.java]: PascalCase.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+	static boolean isSquareFree(int n) {
+
+		if(n%2==0)
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.java]: Spaces around all operators.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+			n=n/2;
+		if(n%2==0)
+			return false;
+		for(int i=3; i<=Math.sqrt(n);i=i+2) {
+
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.java]: Fix indentation.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.java [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+     
+        return true;
+    }
+     
+    public static void main(String[] args)
+    {
+        int n;
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Enter a number");
+        n = sc.nextInt();
```
Comment (arnavb) [inline code/mathematical_algorithms/src/square_free_number/square_free_number.java]: Just declare the variable here.

File: code/mathematical_algorithms/src/square_free_number/square_free_number.c (hunk 2 above)
Comment (mansiag) [reply code/mathematical_algorithms/src/square_free_number/square_free_number.c]: What do you mean?

**Comment by mansiag** [comment]
@arnavb please review.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**
positive_opengenus_cosmos

**Pull request merged**